service/matching/taskWriter.go

```diff
@@ -67,19 +75,31 @@ var errShutdown = errors.New("task list shutting down")
 
 func newTaskWriter(tlMgr *taskListManagerImpl) *taskWriter {
 	return &taskWriter{
-		tlMgr:      tlMgr,
-		config:     tlMgr.config,
-		taskListID: tlMgr.taskListID,
-		stopCh:     make(chan struct{}),
-		appendCh:   make(chan *writeTaskRequest, tlMgr.config.OutstandingTaskAppendsThreshold()),
-		logger:     tlMgr.logger,
+		tlMgr:          tlMgr,
+		db:             tlMgr.db,
+		config:         tlMgr.config,
+		taskListID:     tlMgr.taskListID,
+		taskAckManager: tlMgr.taskAckManager,
+		stopCh:         make(chan struct{}),
+		appendCh:       make(chan *writeTaskRequest, tlMgr.config.OutstandingTaskAppendsThreshold()),
+		logger:         tlMgr.logger,
+		scope:          tlMgr.scope,
 	}
 }
 
-func (w *taskWriter) Start(block taskIDBlock) {
+func (w *taskWriter) Start() error {
```

Comment:
this this called anywhere else? I would just be worried about calling SetAckLevel erroneously

Reply:
its not, so that's all good

Reply:
SetAckLevel doesn't return error
it's called inside start method of tasklistmanager